scripts/uberenv/packages/raja/package.py

```diff
@@ -27,44 +28,33 @@ class Raja(CMakePackage):
     version('0.4.1', tag='v0.4.1', submodules="True")
     version('0.4.0', tag='v0.4.0', submodules="True")
 
-    variant('cuda', default=False, description='Build with CUDA backend')
     variant('openmp', default=True, description='Build OpenMP backend')
 
-    depends_on('cuda', when='+cuda')
-
     depends_on('cmake@3.8:', type='build')
     depends_on('cmake@3.9:', when='+cuda', type='build')
 
-    def _get_sys_type(self, spec):
-        sys_type = spec.architecture
-        # if on llnl systems, we can use the SYS_TYPE
-        if "SYS_TYPE" in env:
-            sys_type = env["SYS_TYPE"]
-        return sys_type
-
     def cmake_args(self):
         spec = self.spec
 
-        sys_type = self._get_sys_type(spec)
-        on_blueos = 'blueos' in sys_type
-        on_blueos_p9 = on_blueos and 'p9' in sys_type
-
         options = []
         options.append('-DENABLE_OPENMP={0}'.format(
             'On' if '+openmp' in spec else 'Off'))
 
         if '+cuda' in spec:
-            if on_blueos_p9:
-                options.extend(['-DCUDA_ARCH=sm_70'])
-            elif on_blueos:
-                options.extend(['-DCUDA_ARCH=sm_60'])
+            if not spec.satisfies('cuda_arch=none'):
+                cuda_arch = spec.variants['cuda_arch'].value
+                options.append('-DCUDA_ARCH=sm_{0}'.format(cuda_arch[0]))
```

Comment:
One quick thing I wanted to mention as a consideration for the future or for whoever maintains RAJA's spack package, is the following:

Since RAJA requires a minumum`sm_35` compute architecture. It may be good to reflect that requirement in the spack package, eg., could one use a `conflicts` statement or something like that?

Reply:
probably is but honestly I'm done making small changes in this PR. If it passes the set of tests I'm running now, it's getting merged...

Reply:
Yeah -- just wanted to bring that up. It may make more sense to log this on Spack's github, since, that's where that is maintained.